File: gitea_repos_master/whatsapp/whatsapp-baileys-api/src/logic/keydb/builder_xml.ts
import { FieldDescription, FieldXMLDescriptor } from 'keydb';

export class KeyDBBuilderXML {
  public readonly description: FieldXMLDescriptor;
  constructor(description: FieldXMLDescriptor) {
    this.description = description;
  }
  public create(field: string, value: any) {
    if (!(field in this.description)) {
      return null;
    }

    const fields: FieldDescription = {
      field,
      value,
      ...this.description[field]
    };
    if (!('description' in fields)) {
      fields.description = fields.title;
    }
    return fields;
  }
}

File: gitea_repos_master/isphere-classes/worker-classes/worker_classes/keydb/builder_xml.py
from typing import Optional

from worker_classes.keydb.interfaces import (
    FieldXMLDescriptor,
    FieldDescription,
    XMLValueType,
)


class KeyDBBuilderXML:
    def __init__(self, description: FieldXMLDescriptor):
        self._description = description

    def create(self, field: str, value: XMLValueType) -> Optional[FieldDescription]:
        if field not in self.description:
            return None

        fields = {"field": field, "value": value, **self.description[field]}
        if "description" not in fields:
            fields["description"] = fields["title"]

        return FieldDescription(**fields)

    @property
    def description(self) -> FieldXMLDescriptor:
        return self._description
